internal/app/server/open_saves_test.go

```diff
@@ -99,6 +112,31 @@ func TestOpenSaves_HealthCheck(t *testing.T) {
 	if want := healthgrpc.HealthCheckResponse_SERVING; got.Status != want {
 		t.Errorf("hc.Check got: %v, want: %v", got.Status, want)
 	}
+	cancel()
+}
+
+func TestOpenSaves_RunServer(t *testing.T) {
+	serviceConfig, err := getServiceConfig()
+	if err != nil {
+		t.Fatalf("getServiceConfig err: %v", err)
+	}
+	ctx := context.Background()
+	t.Run("cancel_context", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+		defer cancel()
+		if err := Run(ctx, "tcp", serviceConfig); err != nil {
+			t.Errorf("got err calling server.Run: %v", err)
+		}
+	})
+	t.Run("sigint", func(t *testing.T) {
+		go func() {
+			time.Sleep(2 * time.Second)
+			syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+		}()
+		if err := Run(ctx, "tcp", serviceConfig); err != nil {
```

Comment:
Do you think it's feasible to test the status change from `SERVING` to `NOT_SERVING`? Looks like `GracefulStop` doesn't return until all RPC calls are served, so timing wise it's a bit tricky.

Reply:
I could, but I added the test to `TestOpenSaves_HealthCheck`